File: src/main/java/net/hectus/neobb/turn/legacy_game/warp/LTVoidWarp.java
package net.hectus.neobb.turn.legacy_game.warp;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.structure.PlacedStructure;
import net.hectus.neobb.turn.default_game.attributes.clazz.Clazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.ColdClazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.NeutralClazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.RedstoneClazz;
import net.hectus.neobb.util.Modifiers;
import org.bukkit.World;
import org.bukkit.potion.PotionEffectType;

import java.util.List;

public class LTVoidWarp extends LWarpTurn {
    public LTVoidWarp(World world) { super(world, "void"); }
    public LTVoidWarp(PlacedStructure data, World world, NeoPlayer player) { super(data, world, player, "void"); }

    @Override
    public void apply() {
        super.apply();
        new Buff.Luck(Buff.BuffTarget.YOU, 15);
        new Buff.Effect(Buff.BuffTarget.OPPONENTS, PotionEffectType.BLINDNESS);
        player.opponents(true).forEach(p -> p.addModifier(Modifiers.P_NO_JUMP));
    }

    @Override
    public int chance() {
        return 30;
    }

    @Override
    public List<Class<? extends Clazz>> allows() {
        return List.of(NeutralClazz.class, ColdClazz.class, RedstoneClazz.class);
    }

    @Override
    public int cost() {
        return 4;
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/legacy_game/warp/LTVoidWarp.kt
package net.hectus.neobb.turn.legacy_game.warp

import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.Effect
import net.hectus.neobb.buff.Luck
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.structure.PlacedStructure
import net.hectus.neobb.turn.default_game.attribute.clazz.Clazz
import net.hectus.neobb.turn.default_game.attribute.clazz.ColdClazz
import net.hectus.neobb.turn.default_game.attribute.clazz.NeutralClazz
import net.hectus.neobb.turn.default_game.attribute.clazz.RedstoneClazz
import net.hectus.neobb.util.Modifiers
import org.bukkit.potion.PotionEffectType
import kotlin.reflect.KClass

class LTVoidWarp(data: PlacedStructure?, player: NeoPlayer?) : LWarpTurn(data, "void", player) {
    override val cost: Int = 4
    override val chance: Double = 30.0
    override val allows: List<KClass<out Clazz>> = listOf(NeutralClazz::class, ColdClazz::class, RedstoneClazz::class)
    override val temperature: Temperature = Temperature.COLD

    override fun apply() {
        Luck(15).apply(player!!)
        Effect(PotionEffectType.BLINDNESS, target = Buff.BuffTarget.OPPONENTS)
        player.opponents().forEach { it.addModifier(Modifiers.Player.NO_JUMP) }
    }
}
